pyproject.toml

```diff
@@ -1,48 +1,179 @@
 [build-system]
 requires = [
-    "setuptools>=42.0",
-    "setuptools_scm[toml]>=3.4"
+    "hatchling>=1.4.1",
+    "hatch-vcs",
 ]
-build-backend = "setuptools.build_meta"
+build-backend = "hatchling.build"
 
-[tool.setuptools_scm]
-write_to = "src/vector/version.py"
+[project]
+name = "vector"
+description = "Vector classes and utilities"
+readme = { file = "README.md", content-type = "text/markdown" }
+maintainers = [ {name = "The Scikit-HEP admins", email = "scikit-hep-admins@googlegroups.com"} ]
+license = "BSD-3-Clause"
+requires-python = ">=3.6"
+keywords = [
+    "vector",
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Information Analysis",
+    "Topic :: Scientific/Engineering :: Mathematics",
+    "Topic :: Scientific/Engineering :: Physics",
+    "Typing :: Typed",
+]
+dependencies = [
+    "importlib-metadata>=0.22;python_version<\"3.8\"",
+    "numpy>=1.13.3",
+    "packaging>=19.0",
+    "typing-extensions;python_version<\"3.8\"",
+]
+dynamic = [
+    "version",
+]
+
+[[project.authors]]
+name = "Jim Pivarski, Henry Schreiner, Eduardo Rodrigues"
+email = "eduardo.rodrigues@cern.ch"
+
+[project.optional-dependencies]
+all = [
```

Comment:
I'm beginning to recommend dropping `all`, it doesn't work well with static config and isn't really useful, IMO; you really shouldn't be installing docs and test and style and everything else all in one venv, increases chances of conflicts.